roles/keystone/templates/etc/keystone/keystone.conf

```diff
@@ -15,9 +17,13 @@ default_log_levels=keystonemiddleware=INFO
 notification_format = cadf
 notification_driver = log
 
-[sql]
+[database]
 connection=mysql+pymysql://keystone:{{ secrets.db_password }}@{{ endpoints.db }}/keystone?charset=utf8
 
+[resource]
+admin_project_name = {{ keystone.admin_project_name }}
+admin_project_domain_name = {{ keysotne.admin_project_domain_name }}
```

Comment:
Also, regarding the heat is_admin_project change - we went to a lot of effort there to make sure that the upgrade path worked. If you don't have these set is_admin_project defaults true and heat will still work - but this is better if it works.